packages/seacas/libraries/ioss/src/Ioss_ElementPermutation.h

```diff
@@ -0,0 +1,283 @@
+// Copyright(C) 1999-2022 National Technology & Engineering Solutions
+// of Sandia, LLC (NTESS).  Under the terms of Contract DE-NA0003525 with
+// NTESS, the U.S. Government retains certain rights in this software.
+//
+// See packages/seacas/LICENSE for details
+
+#pragma once
+
+#include <Ioss_CodeTypes.h>
+#include <Ioss_Utils.h>
+#include <map>    // for map, map<>::value_compare
+#include <string> // for string, operator<
+#include <vector> // for vector
+#include <assert.h>
+
+namespace Ioss {
+  class ElementTopology;
+  class ElementPermutation;
+} // namespace Ioss
+
+namespace Ioss {
+
+using Ordinal = uint16_t;
+using Permutation = uint8_t;
+
+static constexpr Ordinal InvalidOrdinal = 65535;
+static constexpr Permutation InvalidPermutation = 128;
+
+using ElementPermutationMap = std::map<std::string, ElementPermutation *, std::less<std::string>>;
+using EPM_VP                = ElementPermutationMap::value_type;
+
+class EPRegistry
+{
+public:
+  void                            insert(const Ioss::EPM_VP &value, bool delete_me);
+  ElementPermutationMap::iterator begin() { return m_registry.begin(); }
+  ElementPermutationMap::iterator end() { return m_registry.end(); }
+  ElementPermutationMap::iterator find(const std::string &type) { return m_registry.find(type); }
+
+  ~EPRegistry();
+
+private:
+  Ioss::ElementPermutationMap             m_registry;
+  std::vector<Ioss::ElementPermutation *> m_deleteThese;
+};
+
+// Permutation data is stored such that the positive permutations are listed first ... the order of the
+// positive permutations within that group is irrelevant. The remaining permutations listed after the
+// positive ones are the negative permutations hence, any permutation index outside of the positive
+// range is a negative permutation. By convention, the first permutation listed matches the default
+// listed in the Exodus manual
+
+class ElementPermutation
+{
+public:
+  ElementPermutation(const ElementPermutation &) = delete;
+  ElementPermutation &operator=(const ElementPermutation &) = delete;
+
+  virtual ~ElementPermutation();
+
+  void alias(const std::string &base, const std::string &syn);
```

Comment:
The aliasing is technically not needed ... it was one of things that I added in order to more closely mimic ElementTopology and since this is such a new capability, I wasn't sure what typical aliases one would use for permutations so I decided to base them off the topology ones. I can always remove the permutation aliasing but I'm not sure if I should keep the aliasing bookkeeping that I put into the topology code that allows you to get all the aliasing for any given topology.

Reply:
Ok, I can remove that code too